### ISSUE TITLE
Transparency doesnt work with wgpu backend

### ISSUE THREAD

**Issue by oknozor** [hecrj/iced#596]
Seems transparency does not work on wgpu, is this normal ? 
I would happily use glow, but fonts are a bit aliased. Is there a workaround this ? 

- `features = ["wgpu", "default_system_font", "debug", "palette" ]`

<a href="https://ibb.co/H73y0tJ"><img src="https://i.ibb.co/p25Tgfm/screenshot-2020-11-05-T17-06-13-0100.png" alt="screenshot-2020-11-05-T17-06-13-0100" border="0"></a>

- `features = ["glow", "glow_default_system_font", "debug", "palette" ]`

<a href="https://ibb.co/2Kj8Kkt"><img src="https://i.ibb.co/ZThHTM8/screenshot-2020-11-05-T17-05-54-0100.png" alt="screenshot-2020-11-05-T17-05-54-0100" border="0"></a>

**Comment by myfreeweb** [comment]
See https://github.com/hecrj/iced/pull/371 / https://github.com/gfx-rs/wgpu/issues/687

tl;dr for now wgpu hardcodes opaque mode, so it needs to be patched. You can use my fork:

```toml
[patch.crates-io]
# patch: transparency (config.composite_alpha_mode = hal::window::CompositeAlphaMode::PREMULTIPLIED)
wgpu-core  = { git = "https://github.com/myfreeweb/wgpu", branch = "v0.6-alpha" }
wgpu-types = { git = "https://github.com/myfreeweb/wgpu", branch = "v0.6-alpha" }
```

**Comment by code-monad** [comment]
> See #371 / [gfx-rs/wgpu#687](https://github.com/gfx-rs/wgpu/issues/687)
> 
> tl;dr for now wgpu hardcodes opaque mode, so it needs to be patched. You can use my fork:
> 
> ```toml
> [patch.crates-io]
> # patch: transparency (config.composite_alpha_mode = hal::window::CompositeAlphaMode::PREMULTIPLIED)
> wgpu-core  = { git = "https://github.com/myfreeweb/wgpu", branch = "v0.6-alpha" }
> wgpu-types = { git = "https://github.com/myfreeweb/wgpu", branch = "v0.6-alpha" }
> ```

I've followed this & used your fork, modified the examples, but the transparency still seems not working on macOS 11.0.1 Big Sur:
```diff
diff --git a/Cargo.toml b/Cargo.toml
index 6221ae4..6bbfb22 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -100,3 +100,8 @@ iced_web = { version = "0.3", path = "web" }
 [package.metadata.docs.rs]
 rustdoc-args = ["--cfg", "docsrs"]
 features = ["image", "svg", "canvas"]
+
+[patch.crates-io]
+# patch: transparency (config.composite_alpha_mode = hal::window::CompositeAlphaMode::PREMULTIPLIED)
+wgpu-core  = { git = "https://github.com/myfreeweb/wgpu", branch = "v0.6-alpha" }
+wgpu-types = { git = "https://github.com/myfreeweb/wgpu", branch = "v0.6-alpha" }
\ No newline at end of file
diff --git a/examples/clock/Cargo.toml b/examples/clock/Cargo.toml
index c6e3237..b2a819a 100644
--- a/examples/clock/Cargo.toml
+++ b/examples/clock/Cargo.toml
@@ -7,4 +7,4 @@ publish = false
 
 [dependencies]
 iced = { path = "../..", features = ["canvas", "tokio", "debug"] }
-chrono = "0.4"
+chrono = "0.4"
\ No newline at end of file
diff --git a/examples/clock/src/main.rs b/examples/clock/src/main.rs
index b317ac0..3a52ff2 100644
--- a/examples/clock/src/main.rs
+++ b/examples/clock/src/main.rs
@@ -1,3 +1,5 @@
+mod style;
+
 use iced::{
     canvas::{self, Cache, Canvas, Cursor, Geometry, LineCap, Path, Stroke},
     executor, time, Application, Color, Command, Container, Element, Length,
@@ -6,6 +8,11 @@ use iced::{
 
 pub fn main() -> iced::Result {
     Clock::run(Settings {
+	window: iced::window::Settings{
+	    decorations: false,
+	    transparent: true,
+	    ..iced::window::Settings::default()
+	},
         antialiasing: true,
         ..Settings::default()
     })
@@ -32,7 +39,7 @@ impl Application for Clock {
                 now: chrono::Local::now(),
                 clock: Default::default(),
             },
-            Command::none(),
+            Command::none()
         )
     }
 
@@ -55,6 +62,10 @@ impl Application for Clock {
         Command::none()
     }
 
+    fn background_color(&self) -> Color {
+	Color::TRANSPARENT
+    }
+
     fn subscription(&self) -> Subscription<Message> {
         time::every(std::time::Duration::from_millis(500))
             .map(|_| Message::Tick(chrono::Local::now()))
@@ -71,6 +82,7 @@ impl Application for Clock {
             .padding(20)
             .center_x()
             .center_y()
+	    //.style(style::Container)
             .into()
     }
 }
@@ -82,7 +94,6 @@ impl canvas::Program<Message> for Clock {
         let clock = self.clock.draw(bounds.size(), |frame| {
             let center = frame.center();
             let radius = frame.width().min(frame.height()) / 2.0;
-
             let background = Path::circle(center, radius);
             frame.fill(&background, Color::from_rgb8(0x12, 0x93, 0xD8));

```
![image](https://user-images.githubusercontent.com/45656526/101757089-b27dbe00-3b11-11eb-8941-adc977cf10fb.png)



But switching to glow works:
```diff
diff --git a/examples/clock/Cargo.toml b/examples/clock/Cargo.toml
index c6e3237..fce87f2 100644
--- a/examples/clock/Cargo.toml
+++ b/examples/clock/Cargo.toml
@@ -6,5 +6,5 @@ edition = "2018"
 publish = false
 
 [dependencies]
-iced = { path = "../..", features = ["canvas", "tokio", "debug"] }
-chrono = "0.4"
+iced = { path = "../..", features = ["glow", "glow_canvas", "tokio", "debug"] }
+chrono = "0.4"
```
![image](https://user-images.githubusercontent.com/45656526/101757168-c88b7e80-3b11-11eb-801b-68762e25b493.png)

**Comment by hecrj** [comment]
Is this still an issue in the latest versions of `wgpu`?

**Comment by unrelentingtech** [comment]
Yes, you can see that https://github.com/gfx-rs/wgpu/issues/687 is still open. I still use a forked wgpu with a one-line patch for this.

**Comment by LoganWalls** [comment]
I'm still having this same issue on macOS with the latest iced and wgpu, although it looks like wgpu merged that PR. Any advice on a fix / workaround?

Thanks!

**Comment by leoshimo** [comment]
I believe I'm hitting this as well - here is a small sample project:
- https://github.com/leoshimo/commandbar_iced/blob/main/src/main.rs

I've set application + children to have transparent background, but seeing a black box being rendered:
<img width="857" alt="Screenshot 2023-08-15 at 2 39 26 PM" src="https://github.com/iced-rs/iced/assets/56844000/959aeed2-69eb-4e3c-99c8-c4c1b83e48bc">

**Comment by matze** [comment]
Just another data point, on my system transparency works for `WGPU_BACKEND=gl` but not `WGPU_BACKEND=vulkan`.

**Comment by davawen** [comment]
It seems the linked PR has been closed, so perhaps a fix can be merged ?

**Comment by samuelselleck** [comment]
Hi!

Transparency isn't supported on the WebGL backend. Does it make sense to open this as a separate issue?
